weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ObjectVersionsPage.tsx

```diff
@@ -154,65 +172,126 @@ const ObjectVersionsTable: React.FC<{
-        }
-        return null;
-      },
-    }),
-    basicField('createdAtMs', 'Created', {
-      width: 100,
-      renderCell: cellParams => {
-        const createdAtMs = cellParams.value;
-        return <Timestamp value={createdAtMs / 1000} format="relative" />;
-      },
-    }),
-    ...(props.usingLatestFilter
-      ? [
-          basicField('peerVersions', 'Versions', {
-            width: 100,
-            sortable: false,
-            filterable: false,
-            renderCell: cellParams => {
-              const obj: ObjectVersionSchema = cellParams.row.obj;
-              return <PeerVersionsLink obj={obj} />;
-            },
-          }),
-        ]
-      : []),
-  ];
-  const columnGroupingModel: GridColumnGroupingModel = [];
+
+  // TODO: We should make this page very robust similar to the CallsTable page.
+  // We will want to do nearly all the same things: URL state management,
+  // sorting, filtering, ref expansion, etc... A lot of common logic should be
+  // extracted and shared.
+  const {cols: columns, groups: columnGroupingModel} = useMemo(() => {
+    let groups: GridColumnGroupingModel = [];
+    const cols: GridColDef[] = [
+      basicField('object', 'Object', {
+        hideable: false,
+        renderCell: cellParams => {
+          // Icon to indicate navigation to the object version
+          const obj: ObjectVersionSchema = cellParams.row.obj;
+          return (
+            <ObjectVersionLink
+              entityName={obj.entity}
+              projectName={obj.project}
+              objectName={obj.objectId}
+              version={obj.versionHash}
+              versionIndex={obj.versionIndex}
+              fullWidth={true}
+            />
+          );
+        },
+      }),
+    ];
+
+    cols.push(
+      basicField('baseObjectClass', 'Category', {
+        width: 100,
+        valueGetter: cellParams => {
+          return cellParams.row.obj.baseObjectClass;
+        },
+        renderCell: cellParams => {
+          const category = cellParams.value;
+          if (category === 'Model' || category === 'Dataset') {
+            return <TypeVersionCategoryChip baseObjectClass={category} />;
+          }
+          return null;
+        },
+      })
+    );
+
+    if (!props.usingLatestFilter) {
+      const dynamicFields: string[] = [];
+      const dynamicFieldSet = new Set<string>();
+      rows.forEach(r => {
+        Object.keys(r.obj.val).forEach(k => {
+          if (!dynamicFieldSet.has(k)) {
+            dynamicFieldSet.add(k);
+            dynamicFields.push(k);
+          }
+        });
+      });
+
+      const {cols: newCols, groupingModel} =
+        buildDynamicColumns<ObjectVersionSchema>(dynamicFields, (row, key) => {
+          const obj: ObjectVersionSchema = (row as any).obj;
+          const res = obj.val?.[key];
+          if (isTableRef(res)) {
+            // This whole block is a hack to make the table ref clickable. This
```

Comment:
At least in the case of datasets the `rows` column is kind of weird - the view you get today is basically the same as the page for the dataset version overall, just worse because everything is indented.

Reply:
Great call - I hid this column for datasets